src/server/getters/get-profit-loss.ts

```diff
@@ -60,14 +64,29 @@ export interface OutcomeValueTimeseries extends Timestamped {
   transactionHash: string;
 }
 
-export interface ProfitLossResult extends Timestamped, FrozenFunds {
-  netPosition: BigNumber;
-  averagePrice: BigNumber;
-  realized: BigNumber;
-  unrealized: BigNumber;
-  outcome: number;
-  marketId: Address;
-  total: BigNumber;
+// ProfitLossResult is the profit or loss result, at a particular point in
+// time, of a user's position in a single market outcome. ProfitLossResult
+// represents the total accumulation of history (for this market
+// outcome) as of an instantaneous point in time, like a balance sheet in
+// accounting. A user's "position" refers to the shares they have bought
+// ("long" position) or sold ("short" position) in this market outcome.
+export interface ProfitLossResult extends
+  Timestamped, // profit and loss as of this timestamp
+  FrozenFunds { // funds the user froze to be in this position (see FrozenFunds docs)
+  marketId: Address; // user's position is in this market
+  outcome: number; // user's position is in this market outcome
+  netPosition: BigNumber; // current quantity of shares in user's position for this market outcome. "net" position because if user bought 4 shares and sold 6 shares, netPosition would be -2 shares (ie. 4 - 6 = -2). User is "long" this market outcome (gets paid if this outcome occurs) if netPosition is positive. User is "short" this market outcome (gets paid if this outcome does not occur) if netPosition is negative
+  averagePrice: BigNumber; // denominated in tokens/share. average price user paid for shares in the current open position
+  realized: BigNumber; // realized profit in tokens (eg. ETH) user already got from this market outcome. "realized" means the user bought/sold shares in such a way that the profit is already in the user's wallet
+  unrealized: BigNumber; // unrealized profit in tokens (eg. ETH) user could get from this market outcome. "unrealized" means the profit isn't in the user's wallet yet; the user could close the position to "realize" the profit, but instead is holding onto the shares. Computed using last trade price.
+  total: BigNumber; // total profit in tokens (eg. ETH). Always equal to realized + unrealized
+  unrealizedCost: BigNumber; // denominated in tokens. Cost of shares in netPosition
+  realizedCost: BigNumber; // denominated in tokens. Cumulative cost of shares included in realized profit
+  totalCost: BigNumber; // denominated in tokens. Always equal to unrealizedCost + realizedCost
+  realizedPercent: BigNumber; // realized profit percent (ie. profit/cost)
```

Comment:
Can these be calculated? Have them be functions instead of data on an interface?

Reply:
Yes! Agree we should JIT everything where possible. In this case `ProfitLossResult` is the base type for the (serialized) RPC response type for getUserTradingPositions() et al